### ISSUE TITLE
Unity events can't be bound to persistent objects as they are not local to scenes

### ISSUE THREAD

**Issue by adrienlucbert** [DarkOceanInteractive/project-wendigo#54]
# Issue

## Problem description

I just thought there might be something that's annoying.
Due to this change, I don't know if a trigger box (or any other event trigger) can trigger an event on the player. As if each scene has its own version of the PersistentObjects prefab (which includes the player).
I believe (not tested) that if you bind an event (from a trigger box for instance) to anything that's inside the scene's local version of the PersistentObjects instance, then as you run the scene from the menu and the menu scene's PersistentObjects instance is carried along, overriding each scene's local PersistentObjects instance, events won't be bound to anything anymore.

**TLDR**; it might not be possible to bind an event from the inspector to the player or anything else inside the PersistentObjects object. An (annoying) workaround would be to have a script on each trigger with a method (that would be bound to the event trigger), that would call `Singletons.Main.Player.PlayerBody.GetComponent<whateverscript>().whatevermethod()`
I have to test it out, but I'm pretty sure I'm right on this
I put it here bc it means that if you have to deal with interactable objects and trigger boxes, you might not understand why it works if you play the scene directly and not if you play it from the menu.

### With a custom script

> You have a triggerbox (prefab), when you enter it with the player, you want to lock the player's view on an object
You have the triggerbox in the scene, you create a script that looks like:
```csharp
using UnityEngine;

namespace ProjectWendigo
{
    class CustomTriggerBlaBla : MonoBehaviour
    {
        public void TriggerBlaBla(GameObject target)
        {
            Singletons.Main.Player.PlayerBody
              .GetComponent<LockMovementForward>()
              .LockMovementDirection(target);
        }
    }
}
```

attach this script to the triggerbox, and you would call this in the event:

![image](https://user-images.githubusercontent.com/42178413/174444417-c9a88647-c564-4c7c-a694-0a9aeeaa4d1f.png)

### Other possible option

Instead of carrying along the `PersistentObjects`, I'm questioning the very usefulness of having them persistent at all. They should almost all be present in every scene, but do they need to be persistent? Because if not, we could use the scene-local instances and that would fix this issue. However, we must ensure that data such as the settings and notebook content are carried out between scenes.

**Comment by adrienlucbert** [comment]
After discussion with Jasper, the second possible solution will be accepted later, to avoid merge conflicts (as implementing this solution will add changes to every scene).

**Comment by NAkbulut** [comment]
I think having them persist was primarily useful because of some settings and properties having to be the same across different scenes, so persistent objects would pretty much be a good constraint to have. At the same time, I think the complications aren't worth it at the moment, especially considering our deadline. I'm OK with the second possible solution.
